File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0,None)
        current = head
        while(current is not None):
            next_node = current.next
            current.next = dummy.next
            dummy.next = current
            current= next_node
        return dummy.next
            

File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func reverseList(head *ListNode) *ListNode {
    dummy := &ListNode{Val: 0, Next: nil}
	for (head != nil){
		next := head.Next
		head.Next = dummy.Next
		dummy.Next = head 
		head = next
	}
	return dummy.Next
}
